apps/mobile/src/components/big-action-button.tsx

```diff
@@ -1,12 +1,30 @@
-import { Text, TouchableOpacity } from "react-native";
+import { Text, TouchableOpacity, ActivityIndicator, View } from "react-native";
+import clsx from "clsx";
+
+export function BigActionButton({ onPress, text, disabled = false, loading = false }) {
+  const isDisabled = disabled || loading;
 
-export function BigActionButton({ onPress, text }) {
   return (
     <TouchableOpacity
       onPress={onPress}
-      className="mt-6 w-full items-center rounded-full bg-blue-600 py-6 shadow-lg active:opacity-80"
+      className={clsx(
+        "mt-6 w-full items-center justify-center rounded-full py-6 shadow-lg",
+        isDisabled ? "bg-gray-400" : "bg-blue-600 active:opacity-80"
```

Comment:
Might be good to use the variants solutions like you can find some examples in the ui package